packages/toolkit/src/listenerMiddleware/types.ts

```diff
@@ -581,10 +582,7 @@ export type FallbackAddListenerOptions = {
  */
 
 /** @public */
-export type GuardedType<T> = T extends (
-  x: any,
-  ...args: unknown[]
-) => x is infer T
+export type GuardedType<T> = T extends (x: any, ...args: any[]) => x is infer T
```

Comment:
arguments are contravariant so this type wouldn't have worked for guards with other arguments
![image](https://github.com/reduxjs/redux-toolkit/assets/18308300/af1bfb2d-269a-4d95-91b3-f848c5f484e3)